sdk/core/Azure.Core.TestFramework/src/RecordedVariableOptions.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Azure.Core.TestFramework
+{
+    public class RecordedVariableOptions
+    {
+        private readonly Dictionary<string, string> _secretConnectionStringParameters = new Dictionary<string, string>();
+        private string _sanitizedValue;
+
+        public RecordedVariableOptions HasSecretConnectionStringParameter(string name, SanitizedValue sanitizedValue = default)
+        {
+            _secretConnectionStringParameters[name] = GetStringValue(sanitizedValue);
+            return this;
```

Comment:
fluent.. nice

Reply:
While Krzysztof is not looking...

Reply:
This usage of fluent reminds me of Java's Builder concept. It would have been interesting to use this style of optionality in the SDKs instead of the DTO options classes style we went with.